src/model/WaterHeaterHeatPump.cpp

```diff
@@ -544,6 +546,22 @@ namespace model {
       }
     }
 
+    ComponentType WaterHeaterHeatPump_Impl::componentType() const {
+      return ComponentType::Heating;
+    }
+
+    std::vector<FuelType> WaterHeaterHeatPump_Impl::coolingFuelTypes() const {
+      return {};
+    }
+
+    std::vector<FuelType> WaterHeaterHeatPump_Impl::heatingFuelTypes() const {
+      return {FuelType::Electricity};
+    }
+
+    std::vector<AppGFuelType> WaterHeaterHeatPump_Impl::appGHeatingFuelTypes() const {
+      return {AppGFuelType::HeatPump};  // TODO: openstudio-standards uses Electric
+    }
```

Comment:
This is most likely wrong.

* Do we want an HPWH in a zone affecting the answer for that zone's?
* It should probably go check what it's dxCoil is actually doing no?

Reply:
I'd consider it more of a beneficial load than an HVAC component, so no.

Reply:
Agree, a HPWH in a zone should just be considered a zone load, not a zone heating/cooling piece of HVAC equipment. If a HPWH is on a water loop that serves equipment that is heating a zone, then it's heating and cooling fuel type should be "Electricity."

Reply:
Hum. So ThermalZone will call all of its equipment and check the corresponding method...
PlantLoop would call of its supply side equipment and check the corresponding method...


A WaterHeaterHeatPump has a dXCoil and an WaterHeaterMixed. 

The WaterHeaterHeatPump is listed on the ThermalZone's equipment(). 

The PlantLoop lists the tank. The Tank goes to check the WaterHeaterHeatPump (+ its internal heater + any secondaryPlantLoop)

```ruby
m =  Model.new
hpwh= WaterHeaterHeatPump.new(m)
tank = hpwh.tank.to_WaterHeaterMixed.get
z = ThermalZone.new(m)
p = PlantLoop.new(m)
hpwh.addToThermalZone(z)
p.addSupplyBranchForComponent(tank)


tank.setHeaterMaximumCapacity(845000.0)
tank.setHeaterFuelType("Propane")
# This returns the Electricity from the WaterHeaterHeatPump (containingZoneHVACComponent)
tank.heatingFuelTypes.map(&:valueName)
=> ["Electricity", "Propane"]
tank.setHeaterMaximumCapacity(0.0)
=> true
# Cap is zero, so it only returns the WaterHeaterHeatPump
tank.heatingFuelTypes.map(&:valueName)
=> ["Electricity"]

# But thermal zones also checks the WaterHeaterHeatPump
z.heatingFuelTypes.map(&:valueName)
=> ["Electricity"]
```


How do we deal with this?

Reply:
Water Heater Heat Pumps and their tanks should return empty as a heating or cooling fuel; they are not intended to provide space heating.

Reply:
Well, the HPWH's tank can be placed on the supply side of a PlantLoop, in which case it's heating. So I suppose the tank should check for that. The WaterHeaterHeatPump itself could return empty, so the ThermalZone's methods aren't affected. Sound good?